packages/kbn-generate-csv/types/index.ts

```diff
@@ -0,0 +1,262 @@
+  scroll: {
+    size: number;
+    duration: string;
+  };
+  bom: string;
+  separator: string;
+  maxSizeBytes: number | ByteSizeValue;
+  checkForFormulas: boolean;
+  escapeFormulaValues: boolean;
+  escapeValue: (value: string) => string;
+  includeFrozen: boolean;
+}
+
+export interface JobParams {
+  searchSource: SerializedSearchSourceFields;
+  columns?: string[];
+  browserTimezone?: string;
+}
+
+export interface TaskPayloadCSV {
+  searchSource: SerializedSearchSourceFields;
+  columns?: string[];
+  browserTimezone?: string;
+  headers: string;
+  spaceId?: string;
+  isDeprecated?: boolean;
+  objectType: string;
+  title: string;
+  version: string;
+}
+
+export interface CsvConfig {
+  checkForFormulas: boolean;
+  escapeFormulaValues: boolean;
+  maxSizeBytes: number | ByteSizeValue;
+  useByteOrderMarkEncoding: boolean;
+  scroll: {
+    duration: string;
+    size: number;
+  };
+}
+
+export interface TaskRunResult {
+  content_type: string | null;
+  csv_contains_formulas?: boolean;
+  max_size_reached?: boolean;
+  warnings?: string[];
+  metrics?: TaskRunMetrics;
+  /**
+   * When running a report task we may finish with warnings that were triggered
+   * by an error. We can pass the error code via the task run result to the
+   * task runner so that it can be recorded for telemetry.
+   *
+   * Alternatively, this field can be populated in the event that the task does
+   * not complete in the task runner's error handler.
+   */
+  error_code?: string;
+}
+
+interface CsvFromSavedObjectBase {
+  objectType: 'search';
+}
+
+export interface LocatorParams<P extends SerializableRecord = SerializableRecord> {
+  id: string;
+
+  /**
+   * Kibana version used to create the params
+   */
+  version: string;
+
+  /**
+   * Data to recreate the user's state in the application
+   */
+  params: P;
+}
+
+export type BaseParams = Ensure<
+  {
+    layout?: LayoutParams;
+    objectType: string;
+    title: string;
+    browserTimezone: string; // to format dates in the user's time zone
+    version: string; // to handle any state migrations
+  },
+  SerializableRecord
+>;
+
+/**
+ * Report job parameters that an application must return from its
+ * getSharingData function.
+ */
+export type BaseParamsV2 = BaseParams & {
+  locatorParams: LocatorParams[];
+};
+
+/**
+ * Report job parameters, after they are processed in the request handler.
+ */
+export interface BasePayloadV2 extends BaseParamsV2 {
+  headers: string;
+  spaceId?: string;
+  isDeprecated?: boolean;
+}
+
+/**
+ * Makes title optional, as it can be derived from the saved search object
+ */
+export type JobParamsCsvFromSavedObject = CsvFromSavedObjectBase &
+  Omit<BaseParamsV2, 'title'> & { title?: string };
+
+export type TaskPayloadCsvFromSavedObject = CsvFromSavedObjectBase & BasePayloadV2;
+
+export interface JobParamsDownloadCSV {
+  browserTimezone: string;
+  title: string;
+  searchSource: SerializedSearchSourceFields;
+  columns?: string[];
+}
```

Comment:
This file just needs to export the following:
```
export const CONTENT_TYPE_CSV = 'text/csv';
export const CSV_BOM_CHARS = '\ufeff';
export const UI_SETTINGS_CSV_SEPARATOR = 'csv:separator';
export const UI_SETTINGS_CSV_QUOTE_VALUES = 'csv:quoteValues';
export interface JobParams {
export interface CsvConfig {
export interface TaskRunResult {
```

The other things aren't used / shouldn't be used within this package :)

Reply:
Refinement to my previous comment:
As mentioned elsewhere, it'll help to have a `packages/kbn-reporting/common/constant.ts` file with the constants (CONTENT_TYPE_CSV, etc). We just need a `types.ts` file here, and it can be very minimal:

```typescript
import { ByteSizeValue } from '@kbn/config-schema';
import type { SerializedSearchSourceFields } from '@kbn/data-plugin/public';
import { TaskRunMetrics } from '@kbn/reporting-common';

export interface JobParams {
  searchSource: SerializedSearchSourceFields;
  columns?: string[];
  browserTimezone?: string;
}

export interface CsvConfig {
  checkForFormulas: boolean;
  escapeFormulaValues: boolean;
  maxSizeBytes: number | ByteSizeValue;
  useByteOrderMarkEncoding: boolean;
  scroll: {
    duration: string;
    size: number;
  };
}

export interface TaskRunResult {
  content_type: string | null;
  csv_contains_formulas?: boolean;
  max_size_reached?: boolean;
  warnings?: string[];
  metrics?: TaskRunMetrics;
  /**
   * When running a report task we may finish with warnings that were triggered
   * by an error. We can pass the error code via the task run result to the
   * task runner so that it can be recorded for telemetry.
   *
   * Alternatively, this field can be populated in the event that the task does
   * not complete in the task runner's error handler.
   */
  error_code?: string;
}
```